app/logic/Calendar/ActiveSync/ActiveSyncCalendar.ts

```diff
@@ -15,18 +15,15 @@ const kHalfHour = 30 * 60 * 1000; // milliseconds
 
 export class ActiveSyncCalendar extends Calendar implements ActiveSyncPingable {
   readonly protocol: string = "calendar-activesync";
-  readonly events: ArrayColl<ActiveSyncEvent>;
+  declare readonly account: ActiveSyncAccount;
+  declare readonly events: ArrayColl<ActiveSyncEvent>;
   /** Exchange's calendar can only accept incoming invitations from its inbox */
   readonly canAcceptAnyInvitation = false;
   readonly folderClass = "Calendar";
   protected readonly requestLock = new Lock();
 
-  get account(): ActiveSyncAccount {
-    return this.mainAccount as ActiveSyncAccount;
-  }
-
   get serverID() {
-    return new URL(this.url).searchParams.get("serverID");
+    return new URL(this.account.url).searchParams.get("serverID");
```

Comment:
This change is wrong; each ActiveSyncCalendar has its own serverID. You'll need to port 77ffe7f and update all of the places where the url gets used to use the new serverID instead. (Or I could try to do that separately after I've finished #771.)

Reply:
I've merged 77ffe7f in.
But I don't understand the relation between folderID and serverID. Could you make the necessary change, please? Same in ActiveSyncAddressbook.

Reply:
Fixed in #808